instrumentation/rediscala-1.8/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/rediscala/RediscalaClientTest.java

```diff
@@ -0,0 +1,177 @@
+                  new RedisDispatcher("rediscala.rediscala-client-worker-dispatcher"));
+    }
+  }
+
+  @AfterAll
+  static void tearDown() throws Exception {
+    redisServer.stop();
+    if (system != null) {
+      system.getClass().getMethod("terminate").invoke(system);
+    }
+  }
+
+  @Test
+  void testGetCommand() {
+    Pair<Future<Object>, Future<Option<String>>> result =
+        testing.runWithSpan(
+            "parent",
+            () -> {
+              ByteStringSerializer<String> serializer = ByteStringSerializer$.MODULE$.String();
+              ByteStringDeserializer<String> deserializer =
+                  ByteStringDeserializer$.MODULE$.String();
+              Future<Object> writeFuture =
+                  redisClient.set(
+                      "bar",
+                      "baz",
+                      Option.apply(null),
+                      Option.apply(null),
+                      false,
+                      false,
+                      serializer);
+              Future<Option<String>> valueFuture = redisClient.get("bar", deserializer);
+              return Pair.of(writeFuture, valueFuture);
+            });
+
+    await().atMost(java.time.Duration.ofSeconds(3)).until(() -> result.getLeft().isCompleted());
+    await().atMost(java.time.Duration.ofSeconds(3)).until(() -> result.getRight().isCompleted());
+    assertThat(result.getLeft().value().get().get()).isEqualTo(true);
+    assertThat(result.getRight().value().get().get().get()).isEqualTo("baz");
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent").hasNoParent(),
+                span ->
+                    span.hasName("SET")
+                        .hasKind(CLIENT)
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(DbIncubatingAttributes.DB_SYSTEM, "redis"),
+                            equalTo(DB_OPERATION, "SET")),
+                span ->
+                    span.hasName("GET")
+                        .hasKind(CLIENT)
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(DbIncubatingAttributes.DB_SYSTEM, "redis"),
+                            equalTo(DB_OPERATION, "GET"))));
+  }
+
+  @Test
+  public void testSetCommand() {
+    ByteStringSerializer<String> serializer = ByteStringSerializer$.MODULE$.String();
+
+    Future<Object> value =
+        testing.runWithSpan(
+            "parent",
+            () ->
+                redisClient.set(
+                    "foo",
+                    "bar",
+                    Option.apply(null),
+                    Option.apply(null),
+                    false,
+                    false,
+                    serializer));
+    await().atMost(java.time.Duration.ofSeconds(3)).until(value::isCompleted);
```

Comment:
just wondering whether it would be better to use scala await class like the groovy test does or maybe even consider converting the whole test to scala

Reply:
i think  java class is enough to test the  instrument case